test/Microsoft.AspNet.Mvc.Core.Test/Internal/ResponseContentTypeHelperTest.cs

```diff
@@ -0,0 +1,218 @@
+                        new MediaTypeHeaderValue("text/default"),
+                        "text/default"
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ThrowsExceptionOnNullDefaultContentTypeData))]
+        public void ThrowsExceptionOn_NullDefaultContentType(
+            MediaTypeHeaderValue actionResultContentType,
+            string httpResponseContentType)
+        {
+            // Arrange, Act & Assert
+            var exception = Assert.Throws<ArgumentNullException>(() =>
+            {
+                string resolvedContentType = null;
+                Encoding resolvedContentTypeEncoding = null;
+                ResponseContentTypeHelper.ResolveContentTypeAndEncoding(
+                    actionResultContentType,
+                    httpResponseContentType,
+                    null, // default content type
+                    out resolvedContentType,
+                    out resolvedContentTypeEncoding);
+            });
+        }
+
+        public static TheoryData<MediaTypeHeaderValue, string> ThrowsExceptionOnNullDefaultContentTypeEncodingData
+        {
+            get
+            {
+                return new TheoryData<MediaTypeHeaderValue, string>
+                {
+                    {
+                        null,
+                        null
+                    },
+                    {
+                        null,
+                        "text/default"
+                    },
+                    {
+                        new MediaTypeHeaderValue("text/default"),
+                        null
+                    },
+                    {
+                        new MediaTypeHeaderValue("text/default"),
+                        "text/default"
+                    }
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ThrowsExceptionOnNullDefaultContentTypeEncodingData))]
+        public void ThrowsExceptionOn_NullDefaultContentTypeEncoding(
+            MediaTypeHeaderValue actionResultContentType,
+            string httpResponseContentType)
+        {
+            // Arrange
+            var defaultContentType = MediaTypeHeaderValue.Parse("text/bar; p1=p1-value");
+
+            // Act & Assert
+            var exception = Assert.Throws<ArgumentException>(() =>
+            {
+                string resolvedContentType = null;
+                Encoding resolvedContentTypeEncoding = null;
+                ResponseContentTypeHelper.ResolveContentTypeAndEncoding(
+                    actionResultContentType,
+                    httpResponseContentType,
+                    defaultContentType,
+                    out resolvedContentType,
+                    out resolvedContentTypeEncoding);
+            });
+            Assert.Equal(
+                $"The default content type '{defaultContentType.ToString()}' must have an encoding.",
+                exception.Message);
+        }
+    }
```

Comment:
Why is there 100 lines of test code for these exception messages? This should just be a `Debug.Assert`.